server/test/unit/server/db/test_model.py

```diff
@@ -1,19 +1,232 @@
+
+        # make sure the id wasn't replaced
+        self.assertEquals(helper.id, 'foo')
+
+    def test_set_content(self):
+        class ContentUnitHelper(model.ContentUnit):
+            pass
+        helper = ContentUnitHelper()
+        helper.set_content(self.working_dir, 'apples')
+
+    def test_set_content_bad_source_location(self):
+        """
+        Test that the appropriate exception is raised when set_content
+        is called with a non existent source_location
+        """
+        class ContentUnitHelper(model.ContentUnit):
+            pass
+        helper = ContentUnitHelper()
+        try:
+            helper.set_content(os.path.join(self.working_dir, 'foo'), 'apples')
+            self.fail("Previous call should have raised a PulpCodedException")
+        except PulpCodedException, raised_error:
+            self.assertEquals(raised_error.error_code, error_codes.PLP0036)
+
+    def test_set_content_no_relative_path(self):
+        """
+        Test that the appropriate exception is raised when set_content
+        is called with None for the relative_path
+        """
+        class ContentUnitHelper(model.ContentUnit):
+            pass
+        helper = ContentUnitHelper()
+        try:
+            helper.set_content(self.working_dir, None)
+            self.fail("Previous call should have raised a PulpCodedException")
+        except PulpCodedException, raised_error:
+            self.assertEquals(raised_error.error_code, error_codes.PLP0037)
+
+    def test_set_content_empty_relative_path(self):
+        """
+        Test that the appropriate exception is raised when set_content
+        is called with an empty relative_path
+        """
+        class ContentUnitHelper(model.ContentUnit):
+            pass
+        helper = ContentUnitHelper()
+        try:
+            helper.set_content(self.working_dir, '  ')
+            self.fail("Previous call should have raised a PulpCodedException")
+        except PulpCodedException, raised_error:
+            self.assertEquals(raised_error.error_code, error_codes.PLP0037)
+
+    def test_post_save_signal_no_content(self):
+        """
+        Check to make sure that things pass through with no work or errors if no content
+        is set
+        """
+        class ContentUnitHelper(model.ContentUnit):
+            pass
+        foo = ContentUnitHelper()
+        foo._source_location = None
+        foo._relative_path = None
+        model.ContentUnit.post_save_signal(object(), foo)
+
+    @patch('pulp.server.db.model.config')
+    def test_post_save_signal_directory_content(self, mock_config):
+        source_dir = os.path.join(self.working_dir, 'src')
+        target_dir = os.path.join(self.working_dir, 'target')
+        os.mkdir(source_dir)
+        os.mkdir(target_dir)
+        mock_config.config.get.return_value = target_dir
+        # create something in the source directory to test the copy
+        touch(os.path.join(source_dir, 'foo', 'bar'))
```

Comment:
Can we avoid using the pulp utility helpers. There must be a standard library way to do this easily.